spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/web/H2ServerConfiguration.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2015-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.server.config.web;
+
+import java.sql.SQLException;
+
+import javax.servlet.ServletContext;
+
+import org.h2.tools.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.web.servlet.ServletContextInitializer;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.event.ContextClosedEvent;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author Wirth Michael
+ */
+@Configuration
+@ConditionalOnClass(Server.class)
+@ConditionalOnWebApplication
+public class H2ServerConfiguration
+		implements ServletContextInitializer, ApplicationListener<ContextClosedEvent> {
+
+	private static final Logger logger = LoggerFactory
+			.getLogger(H2ServerConfiguration.class);
+
+	@Value("${spring.datasource.url:#{null}}")
+	private String dataSourceUrl;
+
+	private Server server = null;
+
+	public Server initH2TCPServer() {
+		logger.info("Starting H2 Server with URL: " + dataSourceUrl);
+		try {
+			this.server = Server.createTcpServer("-ifNotExists", "-tcp",
+					"-tcpAllowOthers", "-tcpPort", getH2Port(dataSourceUrl)).start();
+		}
+		catch (SQLException e) {
+			throw new IllegalStateException(e);
+		}
+		return server;
+	}
+
+	private String getH2Port(String url) {
+		String[] tokens = StringUtils.tokenizeToStringArray(url, ":");
+		Assert.isTrue(tokens.length >= 5, "URL not properly formatted");
+		return tokens[4].substring(0, tokens[4].indexOf("/"));
+	}
+
+	@Override
+	public void onStartup(ServletContext servletContext) {
+		if (StringUtils.hasText(dataSourceUrl)
```

Comment:
I agree the initialization of the H2Server requires a redesign. The related tests aren't working properly because the test requires to be run in WebApplication to start properly.

Reply:
> The related tests aren't working properly because the test requires to be run in WebApplication to start properly.

Are they failing?

Reply:
I see - the test that you linked in previous comment is passing but not verifying anything.